File: SecondYear/ApoorvaSri123_ApoorvaSrivastava_2226it1116_2/Week_4/second_largest_element.c
#include <stdio.h>
#include <limits.h>

int findSecondLargest(int arr[], int n) {
    if (n < 2) {
        return -1;
    }

    int largest = INT_MIN;
    int secondLargest = INT_MIN;

    for (int i = 0; i < n; i++) {
        if (arr[i] > largest) {
            secondLargest = largest;
            largest = arr[i];
        } else if (arr[i] > secondLargest && arr[i] < largest) {
            secondLargest = arr[i];
        }
    }

    if (secondLargest == INT_MIN) {
        return -1;
    }

    return secondLargest;
}

int main() {
    int numTests;
    scanf("%d", &numTests);
    getchar(); // Consume the newline character

    int results[numTests];

    for (int t = 1; t <= numTests; t++) {
        int n;
        scanf("%d", &n);
        getchar(); // Consume the newline character

        int arr[n];

        for (int i = 0; i < n; i++) {
            scanf("%d", &arr[i]);
        }

        int secondLargest = findSecondLargest(arr, n);
        results[t - 1] = secondLargest;
    }

    for (int t = 1; t <= numTests; t++) {
        printf("%d\n", results[t - 1]);
    }

    return 0;
}

File: SecondYear/praveer7398_PraveerNandan_2226ce1011_2/Week_4/second_largest_element.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class second_largest_element {

    public static int findSecondLargest(int[] arr) {
        if (arr.length < 2) {
            return -1;
        }

        int largest = Integer.MIN_VALUE;
        int secondLargest = Integer.MIN_VALUE;

        for (int num : arr) {
            if (num > largest) {
                secondLargest = largest;
                largest = num;
            } else if (num > secondLargest && num < largest) {
                secondLargest = num;
            }
        }

        if (secondLargest == Integer.MIN_VALUE) {
            return -1;
        }

        return secondLargest;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int numTests = scanner.nextInt();
        scanner.nextLine();

        List<Integer> results = new ArrayList<>();

        for (int t = 1; t <= numTests; t++) {
            int n = scanner.nextInt();
            int[] arr = new int[n];
            scanner.nextLine();

            for (int i = 0; i < n; i++) {
                arr[i] = scanner.nextInt();
            }

            int secondLargest = findSecondLargest(arr);
            results.add(secondLargest);
        }

        for (int t = 1; t <= numTests; t++) {
            System.out.println(results.get(t - 1));
        }

        scanner.close();
    }
}
